src/app/rosetta/lib/account.ml

```diff
@@ -91,304 +80,66 @@ FROM (
-                ) AS nonces GROUP BY nonces.pk_id
-  )
 )
-AS combo GROUP BY combo.pk_id
-|sql} max_txns)
-
-    let query_canonical_fallback =
-      Caqti_request.find_opt
-        Caqti_type.(tup2 string int64)
-        Caqti_type.(tup3 int64 int64 (option int64))
-        {sql| /* See comments on the above query to help understand this one.
-               * Since this query acts on only canonical blocks, we can skip the
-               * recursive traversal part. */
-              SELECT DISTINCT
-                MIN(combo.block_global_slot_since_genesis) AS block_global_slot_since_genesis,
-                MIN(combo.balance) AS balance,
-                MIN(combo.nonce) AS nonce
-              FROM (
-                (SELECT pks.id as pk_id,b.global_slot_since_genesis AS block_global_slot_since_genesis,balance,NULL AS nonce
-
-                FROM blocks b
-                INNER JOIN balances             bal  ON b.id = bal.block_id
-                INNER JOIN public_keys          pks  ON bal.public_key_id = pks.id
-
-                WHERE pks.value = $1
-                AND b.height <= $2
-                AND b.chain_status = 'canonical'
-
-                ORDER BY (bal.block_height, bal.block_sequence_no, bal.block_secondary_sequence_no) DESC
-                LIMIT 1)
-                UNION ALL
-                (SELECT pks.id,NULL,NULL,cmds.nonce
-
-                FROM blocks b
-                INNER JOIN blocks_user_commands busc ON busc.block_id = b.id
-                INNER JOIN user_commands        cmds ON cmds.id = busc.user_command_id
-                INNER JOIN public_keys          pks  ON pks.id = cmds.source_id
-
-                WHERE pks.value = $1
-                AND b.height <= $2
-                AND b.chain_status = 'canonical'
-
-                ORDER BY (b.height, busc.sequence_no) DESC
-                LIMIT 1)
-                )
-              AS combo GROUP BY combo.pk_id
-              |sql}
+AS combo
+|sql}
 
     let run (module Conn : Caqti_async.CONNECTION) requested_block_height
         address =
       let open Deferred.Result.Let_syntax in
       let%bind has_canonical_height = Sql.Block.run_has_canonical_height (module Conn) ~height:requested_block_height in
       if has_canonical_height then (
         match%bind Conn.find_opt query_canonical (address, requested_block_height) with
-        | Some ((_pk,slot,balance,Some nonce)) ->
-            return @@ Some (slot, balance, Some nonce)
-        | Some (_,_,_,None)
+        | Some ((_pk,slot,balance,nonce)) ->
+            return @@ Some (slot, balance, nonce)
         | None ->
-          let%map result = Conn.find_opt query_canonical_fallback (address, requested_block_height) in
-          (* The nonce is returned from this user-command, so we need to add one from here to get the current nonce in the account. *)
-          Option.map result ~f:(fun (slot,balance,nonce) -> (slot,balance,Option.map ~f:Int64.((+) one) nonce)))
+          return @@ None)
```

Comment:
The `@@` operator is redundant here. Besides I can see some code duplication. I wish we could just not `SELECT` `_pk` if it's not really used. But I guess we must `SELECT` it in order for `GROUP BY` to work, right?
If that's the case, I propose to define a local function:
```
let ignore_pk (_pk, slot, balance, nonce) = (slot, balance, none) in
```
And then:
```
let%map row = Conn.find_opt query_pending (address, requested_block_height) in
Option.map ~f:ignore_pk row
```
or something along those lines. And similarily below, reusing `ignore_pk` function.

Reply:
Actually, we don't `GROUP BY` anymore since we are not looking at multiple transactions anymore but simply at a single record on the `accounts_accessed` table, therefore we can remove the `pk_id` field from the query and the code that ignores it.

Reply:
I further simplified the queries to remove redundant code.

Reply:
Great. :) Those `match`es don't seem to do anything anymore, though. Could we get rid of them too?

Reply:
Indeed :)

Reply:
This can be simplified further to:
```
Conn.find_opt
  (if has_canonical_height then query_canonical else query_pending)
  (address, requested_block_height)
```
If you like. I think I like it. :)

Reply:
But I think it's good enough already. :+1: